src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -20,7 +20,14 @@ namespace raylet {
 /// Work represents all the information needed to make a scheduling decision.
 /// This includes the task, the information we need to communicate to
 /// dispatch/spillback and the callback to trigger it.
-typedef std::tuple<Task, rpc::RequestWorkerLeaseReply *, std::function<void(void)>> Work;
+struct Work {
+  Task task;
+  rpc::RequestWorkerLeaseReply *reply;
+  std::function<void(void)> callback;
+  bool waiting_worker_popped;
+  bool dispatch_finished;
+  bool canceled;
+};
```

Comment:
Could we disallow copy and assign on this struct?

Reply:
Good for me.